rfcs/0114-nixos-code-of-conduct.md

```diff
@@ -0,0 +1,112 @@
+be disruptive. A well-defined Code of Conduct gives an explicit
+set of expectations for all contributors, which will also make it easier to identify
+when behavior becomes disruptive.
+
+# Detailed design
+[design]: #detailed-design
+
+These statutes are taken from [rust-lang's Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),
+which provides a great compromise between explicit behaviors and subjective goals.
+Only minor changes have been made.
+
+- We are committed to providing a friendly, safe and welcoming environment for
+all, regardless of level of experience, gender identity and expression,
+sexual orientation, disability, personal appearance, body size, race, ethnicity, age,
+religion (or lack thereof), socioeconomic status, nationality, or other similar characteristics.
+- Please avoid using overtly sexual aliases or other nicknames that might
+detract from a friendly, safe and welcoming environment for all.
+- Please be kind and courteous. There’s no need to be mean or rude.
+- Respect that people have differences of opinion and that every design or
+implementation choice carries a trade-off and numerous costs. There is seldom a single right answer.
+- Please keep unstructured critique to a minimum. If you have solid ideas
+you want to experiment with, make a fork and see how it works.
+- We will exclude you from interaction if you insult, demean or harass anyone.
+That is not welcome behavior. We interpret the term “harassment” as including the definition in the
+[Citizen Code of Conduct](https://github.com/stumpsyn/policies/blob/b8ee50e7a9fc79c39c936830901f8863d493b7cd/citizen_code_of_conduct.md);
+if you have any lack of clarity about what might be included in that concept,
+please read their definition. In particular, we don’t tolerate behavior that excludes
+people in socially marginalized groups.
+- Private harassment is also unacceptable. No matter who you are, if you feel
+you have been or are being harassed or made uncomfortable by a community member,
+please contact one of the channel ops or any of the NixOS moderation team immediately.
+Whether you’re a regular contributor or a newcomer, we care about making this community
+a safe place for you and we’ve got your back.
+- Likewise any spamming, trolling, flaming, baiting or other attention-stealing behavior is not welcome.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+There have been a few spectacular failures of communication and good faith within the community.
+Most notable recent examples include [ad-hominem attacks in a PR review](https://github.com/NixOS/nixpkgs/pull/120729#discussion_r621885348)
+and associated [block evasion discourse thread](https://discourse.nixos.org/t/github-block-evasion-is-not-acceptable/12763),
+heated discussions in [RFC #98](https://github.com/NixOS/rfcs/pull/98) and [RFC #111](https://github.com/NixOS/rfcs/pull/111).
```

Comment:
```suggestion
Most notable recent examples include ad-hominem attacks in a PR review
and an associated Discourse thread about block evasion,
and heated discussions in RFC 98 and RFC 111.
```

I would prefer not to link to that evidence, I made the mistake of taking a look and regretted it instantly.

Reply:
It was recent for some of the items over a year ago, not so recent now. The purpose of the section was to show that there has been issues for years within the nix community. The examples are also cases where I think everyone believed they were doing the "right" thing, and yet there was still fighting. It demonstrates an escalation of community activity because there was little moderation capacity in place.